perllib/FixMyStreet/App/Controller/Admin.pm

```diff
@@ -1916,6 +1943,25 @@ sub ban_user : Private {
     return 1;
 }
 
+sub update_alerts : Private {
+    my ($self, $c) = @_;
+
+    my $changes;
+    for my $alert ( $c->stash->{user}->alerts ) {
+        if ( $c->get_param('delete_alert[' . $alert->id . ']') ) {
+            $alert->delete;
+            $changes = 1;
+        } elsif ( $c->get_param('disable_alert[' . $alert->id . ']') ) {
+            $alert->update({ whendisabled => \"now()" });
+            $changes = 1;
+        } elsif ( $c->get_param('enable_alert[' . $alert->id . ']') ) {
+            $alert->update({ whendisabled => undef });
```

Comment:
Also has a `confirm` function which you might want to use here (I guess you'll want to confirm the alert if you're reenabling it and it isn't, otherwise there's no way to do so).

Reply:
I think so, I imagine someone not getting confirmation email and contacting support is a thing that could happen.